os/arch/arm/src/imxrt/imxrt_pit.c

```diff
@@ -1,153 +1,605 @@
+ * Code
+ ******************************************************************************/
+static uint32_t imxrt_pit_getinstance(PIT_Type *base)
+{
+	uint32_t instance;
+
+	/* Find the instance index from base address mappings. */
+	for (instance = 0; instance < ARRAY_SIZE(s_pitBases); instance++) {
+		if (s_pitBases[instance] == base) {
+			break;
+		}
+	}
+
+	assert(instance < ARRAY_SIZE(s_pitBases));
+
+	return instance;
+}
+
+/*!
+ * brief Ungates the PIT clock, enables the PIT module, and configures the peripheral for basic operations.
+ *
+ * note This API should be called at the beginning of the application using the PIT driver.
+ *
+ * param base   PIT peripheral base address
+ * param config Pointer to the user's PIT config structure
+ */
+void imxrt_pit_init(PIT_Type *base, const pit_config_t *config)
+{
+	assert(config);
+
+#if !(defined(FSL_SDK_DISABLE_DRIVER_CLOCK_CONTROL) && FSL_SDK_DISABLE_DRIVER_CLOCK_CONTROL)
+	/* Ungate the PIT clock*/
+	imxrt_clock_enableclock(s_pitClocks[imxrt_pit_getinstance(base)]);
+#endif /* FSL_SDK_DISABLE_DRIVER_CLOCK_CONTROL */
+
+#if defined(FSL_FEATURE_PIT_HAS_MDIS) && FSL_FEATURE_PIT_HAS_MDIS
+	/* Enable PIT timers */
+	base->MCR &= ~PIT_MCR_MDIS_MASK;
+#endif
+
+#if defined(FSL_FEATURE_PIT_TIMER_COUNT) && (FSL_FEATURE_PIT_TIMER_COUNT)
+	/* Clear the timer enable bit for all channels to make sure the channel's timer is disabled. */
+	for (uint8_t i = 0U; i < FSL_FEATURE_PIT_TIMER_COUNT; i++) {
+		base->CHANNEL[i].TCTRL &= ~PIT_TCTRL_TEN_MASK;
+	}
+#endif /* FSL_FEATURE_PIT_TIMER_COUNT */
+
+	/* Config timer operation when in debug mode */
+	if (config->enableRunInDebug) {
+		base->MCR &= ~PIT_MCR_FRZ_MASK;
+	} else {
+		base->MCR |= PIT_MCR_FRZ_MASK;
+	}
+}
+
+/*!
+ * brief Gates the PIT clock and disables the PIT module.
+ *
+ * param base PIT peripheral base address
+ */
+void imxrt_pit_deinit(PIT_Type *base)
+{
+#if defined(FSL_FEATURE_PIT_HAS_MDIS) && FSL_FEATURE_PIT_HAS_MDIS
+	/* Disable PIT timers */
+	base->MCR |= PIT_MCR_MDIS_MASK;
+#endif
+
+#if !(defined(FSL_SDK_DISABLE_DRIVER_CLOCK_CONTROL) && FSL_SDK_DISABLE_DRIVER_CLOCK_CONTROL)
+	/* Gate the PIT clock*/
+	imxrt_clock_disableclock(s_pitClocks[imxrt_pit_getinstance(base)]);
+#endif /* FSL_SDK_DISABLE_DRIVER_CLOCK_CONTROL */
+}
+
+#if defined(FSL_FEATURE_PIT_HAS_LIFETIME_TIMER) && FSL_FEATURE_PIT_HAS_LIFETIME_TIMER
+
+/*!
+ * brief Start the lifetime counter
+ *
+ * The lifetime timer is a 64-bit timer which chains timer 0 and timer 1 together.
+ * Timer 0 and 1 are chained by calling the PIT_SetTimerChainMode before using this timer.
+ * The period of lifetime timer is equal to the "period of timer 0 * period of timer 1".
+ *
+ * param base PIT peripheral base address
+ */
+static inline void imxrt_pit_start_lifetimetimercount(PIT_Type *base)
+{
+	tmrvdbg("Start lifetime counter\n");
+	base->CHANNEL[kPIT_Chnl_1].LDVAL = 0xFFFFFFFF;
+	base->CHANNEL[kPIT_Chnl_1].TCTRL = 0x0;
+	base->CHANNEL[kPIT_Chnl_1].TCTRL |= PIT_TCTRL_CHN_MASK;
```

Comment:
Need to include `chip/imxrt102x_config.h` and `chip/imxrt105x_config.h`

Reply:
Not needed. This is been included in ./arch/arm/src/imxrt/imxrt_clock.h

Reply:
ok